internal/pubsub/subscriber.go

```diff
@@ -103,167 +90,250 @@ func (s Subscriber) Listen() error {
 }
 
 // listen listens for messages on a channel and calls the registered message handler
-func (s Subscriber) listen(messages <-chan *message.Message, wg *sync.WaitGroup) {
+func (s Subscriber) listen(messages <-chan events.Request[events.AuthRelationshipRequest, events.AuthRelationshipResponse], wg *sync.WaitGroup) {
 	defer wg.Done()
 
 	for msg := range messages {
+		elogger := s.logger.With(
+			"event.message.topic", msg.Topic(),
+			"event.message.action", msg.Message().Action,
+			"event.message.object.id", msg.Message().ObjectID.String(),
+			"event.message.relations", len(msg.Message().Relations),
+		)
+
 		if err := s.processEvent(msg); err != nil {
-			s.logger.Warn("Failed to process msg: ", err)
+			elogger.Errorw("failed to process msg", "error", err)
 
-			msg.Nack()
-		} else {
-			msg.Ack()
+			if nakErr := msg.Nak(nakDelay); nakErr != nil {
+				elogger.Warnw("error occurred while naking", "error", nakErr)
+			}
+		} else if ackErr := msg.Ack(); ackErr != nil {
+			elogger.Errorw("error occurred while acking", "error", ackErr)
```

Comment:
Given the nature of request/reply, do we want to nak these messages at all or to just reply with an error on the first attempt? This seems like a question around who is responsible for retrying if an otherwise good request fails.

Reply:
correct. if an error is produced it's given back to the requestor and they should handle the retries. however as you mentioned in another review, we should retry for reply failures. So this has been left as is, and we now only return an error for reply failures.